app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -14,20 +16,43 @@ def parse_workflow(wf_name, wf_config)
           step_key, step_config = step.first
           cmd_step << translate_step(step_key, step_config)
         end
-
+        process_workflow_envs(wf_config, cmd_step)
         cmd_step
       end
 
+      def process_workflow_envs(wf_config, cmd_step)
+        cmd_step.env ||= {}
+        BK::Compat::Bitrise::EnvHelper.process_env_variables(wf_config['envs'], cmd_step.env)
+      end
+
+      def process_env_variable(env, target_env)
+        env.each do |key, value|
+          next if key == 'opts' || key.to_s.strip.empty? || value.to_s.strip.empty?
+
+          target_env[key] = value
+        end
+      end
+
       def translate_step(step_key, step_config)
         step_inputs = (step_config['inputs'] || {}).reduce({}, :merge)
         load_workflow_step(step_key, step_inputs).tap do |translated_step|
           if translated_step.respond_to?(:agents) && step_config.include?('stack')
             translated_step.agents ||= {}
             translated_step.agents[:stack] = step_config['stack']
           end
+
+          # Apply environment variables if available
+          apply_env_config(translated_step, step_config)
         end
       end
 
+      def apply_env_config(step, config)
+        return unless step.respond_to?(:env) && config['envs']
+
+        step.env ||= {}
+        BK::Compat::Bitrise::EnvHelper.process_env_variables(config['envs'], step.env)
```

Comment:
if `BK::Compat::Bitrise::EnvHelper.process_env_variables` didn't take the second parameter and returned a dictionary (as mentioned in other comments) this should be:

```suggestion
        step.env.merge!(BK::Compat::Bitrise::EnvHelper.process_env_variables(config['envs']))
```